### PR TITLE
bug: document rest api long polling limitation

### LINKED ISSUE #4375: Document limitation of long-polling job activation via REST API
### Description

The Camunda Java client 8.6.0 can perform long polling job activation via the Camunda REST API.

However, due to a limitation of the embedded HTTP client, the Camunda Java client is unable to maintain a long-polling connection via REST API (i.e. HTTP 1.1) when:
* Jobs larger than the maximum message size are sent in batches
* Additional job activation requests/connections are received and closed

See [bug report](https://github.com/camunda/camunda/issues/19883) for more details.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Camunda Java client 8.6.0 can't maintain a long-polling connection under certain conditions. The connection is broken when returning truncated jobs (i.e. when jobs are larger than the maximum message size), or when additional job activation requests are sent and completed. This limitation is documented with this commit.

closes #4375

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by koevskinikola** [comment]
ℹ️ These changes should be available in the `8.6.0` docs. If the `8.6.0` versioned docs are created before this PR is merged, please ping me, so that I can backport the changes.

File: docs/apis-tools/java-client/job-worker.md [hunk 1]
```diff
@@ -13,6 +13,18 @@ keywords: ["backpressure", "back-pressure", "back pressure"]
...
+
+Camunda 8.6.0 introduces job polling via REST API. However, the HTTP client that Camunda Java client 8.6.0
+uses isn't able to maintain a long-living polling connection, i.e. perform long-polling job activation when
+activating jobs larger than the maximum message size, or when additional job activation requests are received
+while the long-polling connection is still open.
+
+If you encounter this issue, consider switching to the Zeebe gRPC protocol for job activation, or use job
+activation via the REST API with long polling disabled.
+
+:::
```
Comment (conceptualshark) [inline docs/apis-tools/java-client/job-worker.md:26]:
```suggestion
:::caution REST API limitation

Job polling via REST API is available in Camunda 8.6.0, but is not suitable for long-lived polling connections (for example, when activating jobs larger than the maximum message size, or when additional job activation requests are received while the long-polling connection is still open).

If you encounter this issue, consider switching to the Zeebe gRPC protocol for job activation, or use job activation via the REST API with long polling disabled.

:::
```

Is this form of the limitation still correct? If explaining the client limitations is necessary, we can leave it as-is - I'd just put the examples in parentheses to make it a bit easier to read.

File: docs/apis-tools/java-client/job-worker.md [hunk 2]
```diff
@@ -13,6 +13,18 @@ keywords: ["backpressure", "back-pressure", "back pressure"]
 
 The Java client provides a job worker that handles polling for available jobs. This allows you to focus on writing code to handle the activated jobs.
 
+:::caution REST API limitation
+
+Camunda 8.6.0 introduces job polling via REST API. However, the HTTP client that Camunda Java client 8.6.0
+uses isn't able to maintain a long-living polling connection, i.e. perform long-polling job activation when
+activating jobs larger than the maximum message size, or when additional job activation requests are received
+while the long-polling connection is still open.
```
Comment (tmetzke) [inline docs/apis-tools/java-client/job-worker.md:21]: > [...] or when additional job activation requests are received while the long-polling connection is still open.

We might need to clarify this a bit more:
* Does this happen when multiple job activation requests come from the same client?
* Does this happen when multiple clients run in the same JVM application?
* Does this happen when multiple clients run independently as separate JVM applications?

File: docs/apis-tools/java-client/job-worker.md (hunk 1 above)
Comment (koevskinikola) [reply docs/apis-tools/java-client/job-worker.md:26]: We can also format the examples as a list. I can expand them with the clarifications that @tmetzke suggested.

Comment (conceptualshark) [reply docs/apis-tools/java-client/job-worker.md:26]: If there is a need for additional examples, I agree the list might be easier to maintain clarity.

**Comment by christinaausley** [comment]
As far as this PR goes for `8.6` readiness, @tmetzke would like to merge a version of the current content that we're fine with from a tech writer's perspective, and we should ship this caution warning with the release. We can improve on the examples once Nikola is back in a follow-up to refine the docs.

@conceptualshark Given you are assigned here, would you like to polish this to merge, or would you like me to take a look? Happy to help!

**Review by christinaausley — DISMISSED** [review]
This looks great @conceptualshark. I just split the first part into two sentences -- thank you!

**Review by tmetzke — APPROVED** [review]
Thanks, folks, this looks really good already 👍 
@koevskinikola, let's quickly follow-up on this when you're back to talk through the examples and make them more specific. We can adjust the docs after release to improve them a bit in this regard.

**Comment by akeller** [comment]
This was not merged in time for the docs release cutting and will need to be backported.

**Comment by christinaausley** [comment]
Meant to update main and totally botched this. I'm just gonna open a new PR to backport this and loop in the feedback here.

**Comment by christinaausley** [comment]
Closing in favor of https://github.com/camunda/camunda-docs/pull/4426.

**Pull request closed without merging**